security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -92,7 +91,8 @@ func createClientset() kubernetes.Interface {
 }
 
 // NewSecretFetcher returns a pointer to a newly constructed SecretFetcher instance.
-func NewSecretFetcher(ingressGatewayAgent bool, endpoint, CAProviderName string, tlsFlag bool, clientSet kubernetes.Interface) (*SecretFetcher, error) {
+func NewSecretFetcher(ingressGatewayAgent bool, endpoint, CAProviderName, namespace string,
```

Comment:
better remove passing ```namespace``` since it's ingress specific, just set ns in this func if ```ingressGatewayAgent==true```

Reply:
According to the design, we don't want to hard code the namespace to watch. It could be istio-system or others, depending on which namespace the ingress gateway is running. @myidpt could you take a look?

Reply:
namespace could be set by getting from env variable in NewSecretFetcher, just don't pass non-general params in this function otherwise input params grows longer and longer when adding more logic..

Reply:
I see. Let me refactor this part. Thanks.

Reply:
thanks !